great_expectations/rule_based_profiler/domain_builder/domain.py

```diff
@@ -0,0 +1,18 @@
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import IDDict
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.types import SerializableDictDot
+
+
+@dataclass
+class Domain(SerializableDictDot):
+    domain_kwargs: Optional[Dict[str, Union[str, Dict[str, Any]]]] = None
+
+    def to_json_dict(self) -> dict:
+        return convert_to_json_serializable(data=asdict(self))
+
+    @property
+    def id(self) -> str:
```

Comment:
I think this should subclass IDDict so that isinstance(domain_kwargs, IDDict) is true

Reply:
@jcampbell Done.